### DISCUSSION TITLE
LC passive filter design simulation and netlist export

### DISCUSSION THREAD

**Discussion by csrabak** [ra3xdh/qucs_s#803] [General]
In present (24.2.0) version, a filter designed using Filter Synthesis (Ctrl-2) and pasted to the schematic canvas does not simulate in Ngspice, in fact it gives errors, and if the simulator is changed to Qucsator it simulates but then the netlist export gives an error message: "This action is supported only for SPICE simulators"!

Is this the proposed design or is this a consquence of the progress on the support to SPICE simulation and this is a corner case?

**Comment by ra3xdh** [comment]
> a filter designed using Filter Synthesis (Ctrl-2) and pasted to the schematic canvas does not simulate in Ngspice

What filter did you designed with synthesizer? The simulation works as expected on my machine. Only LC filters design is possible for Ngspice backend. Ngspice doesn't support microstrips. 

> but then the netlist export gives an error message: 

It is not a bug. The netlist export from the main menu is needed only for SPICE simulator. For Qucsator press F6 -*Simulation->Show last netlist* and then copy the netlist.

**Comment by csrabak** [comment]
I've tried a simple LC filter and as you write, I also got the simulation if the simulator _qucsator_ is chosen.

The steps I went through are as follows:

Specified the filter Ctrl-2:

![image](https://github.com/ra3xdh/qucs_s/assets/10200568/cc8ca568-67f2-4c8a-abd5-ef3737c04fb2)

Pasted in the circuit canvas:

![cauer](https://github.com/ra3xdh/qucs_s/assets/10200568/cdc81120-ce76-4e43-9314-8d81f3394827)

Simulation:

![image](https://github.com/ra3xdh/qucs_s/assets/10200568/1abe19d5-d6b3-4c22-a08f-94ca541458c2)

Checking the netlist with F6 gives:

```
# Qucs 24.2.0  D:/Users/csrabak/cauer.sch

C:C2 _net0 _net1 C="1.38204pF" V=""
L:L1 _net0 _net1 L="593.191nH" I=""
C:C3 gnd _net1 C="18.7018pF" V=""
Eqn:Eqn1 dBS21="dB(S[2,1])" dBS11="dB(S[1,1])" Export="yes"
Pac:P1 _net0 gnd Num="1" Z="200 Ohm" P="0 dBm" f="1 GHz" Temp="26.85"
.SP:SP1 Type="log" Start="7MHz" Stop="700MHz" Points="200" Noise="no" NoiseIP="1" NoiseOP="2" saveCVs="no" saveAll="no"
C:C6 _net2 _net3 C="4.64571pF" V=""
L:L3 _net2 _net3 L="462.777nH" I=""
C:C7 gnd _net3 C="9.52143pF" V=""
Pac:P2 _net3 gnd Num="2" Z="200 Ohm" P="0 dBm" f="1 GHz" Temp="26.85"
C:C4 _net1 _net2 C="6.29691pF" V=""
L:L2 _net1 _net2 L="463.574nH" I=""
C:C5 gnd _net2 C="16.9416pF" V=""
C:C1 gnd _net0 C="12.0435pF" V=""

```

Now, this netlist is not useable for further simulations (the aim here is to use SPICE for other analysis, specially introducing real component models and Monte Carlo), as the two `Pac` circuit elements are unknows to SPICE like simulators.

So, my understanding is that in present development status, this interim netlist has to be get and hand edited in order to be amenable to further analysis.

But lets try changing the simulator:

Changing to Ngspice, the netlist changes to:

```
* Qucs 24.2.0  D:/Users/csrabak/cauer.sch
.INCLUDE "C:/Program Files/Qucs-S/share/qucs-s/xspice_cmlib/include/ngspice_mathfunc.inc"
.PARAM dBS21=dB(S[2,1])
.PARAM dBS11=dB(S[1,1])
C2 _net0 _net1  1.38204P 
L1 _net0 _net1  593.191N 
C3 0 _net1  18.7018P 
VP1 _net0 0 dc 0 ac 0.632456 SIN(0 0.632456 1G) portnum 1 z0 200
C6 _net2 _net3  4.64571P 
L3 _net2 _net3  462.777N 
C7 0 _net3  9.52143P 
VP2 _net3 0 dc 0 ac 0.632456 SIN(0 0.632456 1G) portnum 2 z0 200
C4 _net1 _net2  6.29691P 
L2 _net1 _net2  463.574N 
C5 0 _net2  16.9416P 
C1 0 _net0  12.0435P 

.control

let dBS21=dB(S[2,1])
let dBS11=dB(S[1,1])
SP DEC 100 7MEG 700MEG
write spice4qucs.sp1.plot S_1_1 Y_1_1 Z_1_1 S_1_2 Y_1_2 Z_1_2 S_2_1 Y_2_1 Z_2_1 S_2_2 Y_2_2 Z_2_2
destroy all
reset

exit
.endc
.END
```

Which, reading it in an editor seemed me OK and I found clever that the Pac are replaced by Ngspice's VSRC.

However, the simulation gets back with the following message:

![image](https://github.com/ra3xdh/qucs_s/assets/10200568/4e97b4e6-4e8f-47f7-b712-b39bac57f468)

And, of course no data is available if the data monitor canvas is selected.

As the SW and docs are in a state of flux, am I doing something incorrectly, or do I need to tweak the schematic in order the SPICE simulations work?

**Comment by ra3xdh** [comment]
> I've tried a simple LC filter and as you write, I also got the simulation if the simulator qucsator is chosen.

There is a bug in a filter synthesizer. The netlist export to Ngspice will not help you. The Cauer filter substitutes an invalid equation into the schematic. Only Cauer LC-filter is affected. Othe filter types work as expected. You may remove an equation or replace it by Nutmeg equation and simulation will continue without error. You may synthesize a Butterworth filter and look for valid equation in the synthesized schematic.  I will provide a fix soon.

> Now, this netlist is not useable for further simulations (the aim here is to use SPICE for other analysis, specially introducing real component models and Monte Carlo)

It is Qucsator netlist. Qucsator is not a SPICE simulator and uses another netlist syntax.

**Comment by ra3xdh** [comment]
See #806. I will provide a fix for Cauer filter soon. As mentioned above, you may correct the equations manually.

**Reply by csrabak** [reply]
Will do meanwhile these and other stuff is collected for a new release, and keep a note on my notebook.

Thanks

**Comment by ra3xdh** [comment]
The equation error has been fixed by #807.
